### PR TITLE
werft/config: Delete core-dev namespace on branch delete

### PR DESCRIPTION
**Pull request by ArthurSens**
Signed-off-by: ArthurSens <arthursens2005@gmail.com>

## Description

@mads-hartmann @csweichel I'm opening this one as a draft because I'm not entirely sure this is all that we need to do.

How do we tell wipe-devstaging the namespace to delete?

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/981

## How to test

Also a hard one to test, since this needs to be on main to execute

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by csweichel** [comment]
> How do we tell wipe-devstaging the namespace to delete?

We derive that from the branch the job was triggered on. 
For good measure, we should remove the "wipe all" behaviour from the job to avoid accidentally removing all preview environments.

**Review by mads-hartmann — COMMENTED** [review]
A few early comments (In addition to Chris' suggestion of removing the the "wipe all" feature from the job)

1. Lets rename the job to delete-preview-environment and add ourselves as the CODEOWNER
2. For figuring out the namespace I the information is part of the context (e.g. you can see what such a context looks like based on the dump that's done in the build [here](https://github.com/gitpod-io/gitpod/blob/main/.werft/build.yaml#L206) - e.g. [here's an example](https://werft.gitpod-dev.com/job/gitpod-build-main.2392/raw).

**Comment by ArthurSens** [comment]
> Lets rename the job to delete-preview-environment and add ourselves as the CODEOWNER

I believe we want to create a new job for this... We currently have 38 open pull requests, with some of them are open for a really long time...

I think we want to keep wipe-devstaging as it is today (deleting previews older than 2 days), but create a new job that deletes previews as soon as PRs are merged/branch is destroyed

**Comment by mads-hartmann** [comment]
@ArthurSens feel free to create a new job for this PR ☺️ 

Then we can follow up with wipe-devstaging after. I'm leaning towards removing wipe-devstaging and also sweeper in favour or a nightly Werft job that performs cleanup of old preview environments. But I need to understand the use-case for wipe-devstaging and Sweeper a bit better first.

**Comment by mads-hartmann** [comment]
@ArthurSens with https://github.com/gitpod-io/gitpod/pull/8116 merged I was able to remove the other teams as reviewers :)

File: .werft/delete-preview-environments/delete-preview-environment.yaml
Comment (mads-hartmann) [inline .werft/delete-preview-environments/delete-preview-environment.yaml:53]: I don't believe you need for this job as you're not performing any Leeway builds. If so you can also remove the NODENAME env

File: .werft/delete-preview-environments/delete-preview-environment.yaml
Comment (mads-hartmann) [inline .werft/delete-preview-environments/delete-preview-environment.yaml:60]: You don't need any sidecars for this job I believe

File: .werft/delete-preview-environments/delete-preview-environment.yaml
Comment (mads-hartmann) [inline .werft/delete-preview-environments/delete-preview-environment.yaml:1]: I'd just get rid of this :)

File: .werft/delete-preview-environments/delete-preview-environment.ts
Comment (mads-hartmann) [inline .werft/delete-preview-environments/delete-preview-environment.ts:44]: Are you sure the name of the namespace can be derived by the Job name?

I don't see any delete-preview-* jobs on https://werft.gitpod-dev.com/ - did you run this? :) 

I was expecting you'd have to get the namespace based on Repository.ref but I might be wrong.

File: .werft/delete-preview-environments/delete-preview-environment.yaml
Comment (mads-hartmann) [inline .werft/delete-preview-environments/delete-preview-environment.yaml:59]: You could leave this out of this PR and we could merge it to test it on main. That way we get to see the contents of context.json when branches are deleted to verify it has the information we need to compute the namespace name

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
Left a few suggestions :)

File: .werft/delete-preview-environments/delete-preview-environment.ts
Comment (ArthurSens) [reply .werft/delete-preview-environments/delete-preview-environment.ts:44]: I just copied this part from `build.ts`.

Over there the namespace is being derived from the job name 🤔
I believe because werft already does the name sanitization (get rid of unsupported characters)

**Review by mads-hartmann — APPROVED** [review]

**Pull request merged**
